patroni/postgresql.py

```diff
@@ -1459,21 +1518,49 @@ def clone(self, clone_member):
 
         ret = self.create_replica(clone_member) == 0
         if ret:
-            self._major_version = self.get_major_version()
-            self._server_parameters = self.get_server_parameters(self.config)
-            self.delete_trigger_file()
-            self.restore_configuration_files()
+            self._post_restore()
+            self._configure_server_parameters()
         return ret
 
     def bootstrap(self, config):
         """ Initialize a new node from scratch and start it. """
-        if self._initialize(config) and self.start() and self.run_bootstrap_post_init(config):
-            for name, value in (config.get('users') or {}).items():
-                if name not in (self._superuser.get('username'), self._replication['username']):
-                    self.create_or_update_role(name, value['password'], value.get('options', []))
-            self.create_or_update_role(self._replication['username'], self._replication['password'], ['REPLICATION'])
+        method = config.get('method') or 'initdb'
+        self._running_custom_bootstrap = method != 'initdb' and method in config and 'command' in config[method]
+        if self._running_custom_bootstrap:
+            do_initialize = self._custom_bootstrap
+            config = config[method]
         else:
-            raise PostgresException("Could not bootstrap master PostgreSQL")
+            do_initialize = self._initdb
+        return do_initialize(config) and self._configure_server_parameters() and self.start()
+
+    def post_bootstrap(self, config, task):
+        try:
+            self.create_or_update_role(self._superuser['username'], self._superuser['password'], ['SUPERUSER'])
+
+            # We were doing a custom bootstrap instead of running initdb, therefor we opened trust
+            # access from certain addresses to be able to reach cluster and change password
+            if self._running_custom_bootstrap:
+                self._running_custom_bootstrap = False
+                if not self.config.get('pg_hba'):
```

Comment:
May it is confusing , but `hba_file` and `pg_hba` are different things...
It was possible to define a custom `hba_file` even before.
`pg_hba` - is a custom pg_hba configuration which would be enforced by Patroni.

In this case, if we don't have `pg_hba` defined, we need to restore original file.
On the line 1548 it will write into pg_hba.conf only if we have `pg_hba`.